### ISSUE TITLE
implement "Evaluate ENVI header" algorithm

### ISSUE THREAD

**Issue by janzandr** [EnMAP-Box/enmap-box#148]
Discussed with @aloboa and @jakimowb in #25.

We already can write an ENVI header with all the metadata from the GDAL Data Model. This way, the ENVI software is aware of all the metadata specified inside QGIS / EnMAP-Box.

It is proposed to implement an "Evaluate ENVI header" algorithm to support the opposite direction: evaluating an ENVI header and updating the GDAL Data Model. This way, the ENVI header can be edited externally and the changes can be evaluated afterwards.

Also place a shortcut here:
![image](https://user-images.githubusercontent.com/45030715/194727797-79f4597e-e1e6-4fcd-bdb1-b71aa32e84ee.png)

**Comment by aloboa** [comment]
I suggest that this process be, at least optionally, run upon reading any hdr file, reporting any missing or incorrect values of
those hdr fields that are likely of being required later. Note that at the time of reading, the only strictly required fields are (AFAIK): 
samples
lines
bands 
data type
header offset
file type 
interleave
byte order
(and actually, the last 3 in this list might have default values)

But the fact is that, for real work with hyperspectral images, the following will be, at some time, needed and/or highly useful:
map info
pixel size
x start
y start
band names
wavelength
fwhm
wavelength units
bbl
data gain values
data offset values
data ignore values
data reflectance gain values
data reflectance offset values
default bands
default stretch
(https://www.l3harrisgeospatial.com/docs/enviheaderfiles.html)
I would report the values for all these fields, noting if they have been read from the actual hdr or assumed by default. In that way the user:
1. Can make sure the values read by EnMapBox are the ones featured in the hdr file.
2. Is warned of eventual important fields that are not present in the hdr file.

There are more optional metadata (https://www.l3harrisgeospatial.com/docs/enteroptionalheaderinformation.html#DataReflectanceGainValues)
but I'm not sure these fields are actually ever applied or are just there for completeness. For example, sun angles can be recorded in the hdr file, but atmospheric correction programs will typically ask for these values to be entered and not automatically read.

**Comment by janzandr** [comment]
> I suggest that this process be, at least optionally, run upon reading any hdr file, reporting any missing or incorrect values

I agree, that from an ENVI user perspective, this would be optimal behaviour.

I will discuss this with the team.

**Comment by aloboa** [comment]
> I agree, that from an ENVI user perspective, this would be optimal behaviour.

From an ENVI perspective, this is irrelevant. ENVI users do not use EnMapBox.
This is important for EnMapBox users, as ENVI format is the most common format for hyperspectral imagery.
If you want to facilitate ENVI users to become EnMapBox users, this is even more important.

**Comment by janzandr** [comment]
Yes, thats what I meant :-)
